Docs: add delegation use-case and clean up existing use-cases (#3138)
input-output-hk/cardano-wallet: docs/user-guide/common-use-cases/how-to-make-a-transaction.md

[`POST /byron-wallets/{walletId}/transactions`](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/postByronTransaction) - transaction from **Byron** wallet

Behind the scene, the wallet engine will select necessary inputs from the wallet, generate a change address within the wallet, sign and submit the transaction. A transaction can have multiple outputs, possibly to the same address. Note that in Byron, addresses are necessarily base58-encoded (as an enforced convention).

## New transaction workflow

[New transaction workflow](https://input-output-hk.github.io/cardano-wallet/api/edge/#tag/Transactions-New) decouples creation of the transaction into separate steps:
 - Construct: [`POST /wallets/{walletId}/transactions-construct`](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/constructTransaction)
 - Sign: [`POST /wallets/{walletId}/transactions-sign`](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/signTransaction)
 - Submit: [`POST /wallets/{walletId}/transactions-submit`](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/submitTransaction)

> Behind the scene, on the **construct** step, the wallet engine will select necessary inputs from the wallet, generate a change address within the wallet and calculate the `fee` for this particular transaction.

Reviewer: ```suggestion
Behind the scene, on the **construct** step, the wallet engine will select necessary inputs belonging to the wallet, generate a change address of the wallet and calculate the `fee` for this particular transaction.
```

-H "Content-Type: application/json"

{
  "id": "35dd9db1f61822ac82f4690ea4fe12426bf6e534aff8e13563fce55a4d502772"
}
```

We can monitor status of submitted transaction using [GET /wallets/{walletId}/transactions/{transactionId}](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/getTransaction) endpoint.
```
$ curl -X GET http://localhost:8090/v2/wallets/1b0aa24994b4181e79116c131510f2abf6cdaa4f/transactions/35dd9db1f61822ac82f4690ea4fe12426bf6e534aff8e13563fce55a4d502772 | jq

{
  "inserted_at": {
    "height": {
      "quantity": 2975821,
      "unit": "block"
    },
    "time": "2021-10-08T11:25:32Z",
    "epoch_number": 161,
    "absolute_slot_number": 39323116,
    "slot_number": 140716
  },
  "status": "in_ledger",
....
```

## Transaction history

Note that all transactions made from and to any wallet are available in the transaction history.
We can always display details of a particular transaction as well as list all transactions that are known to a wallet.

> For instance transactions can be tracked via:

Reviewer: ```suggestion
For instance, transactions can be tracked via:
```

[`DELETE /wallets/{walletId}/transactions/{transactionId}`](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/deleteTransaction)  - transactions from **Shelley** wallet
[`DELETE /byron-wallets/{walletId}/transactions/{transactionId}`](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/deleteByronTransaction)  - transactions from **Byron** wallet

For more information about transactions lifecycle, have a look at [[About-Transactions-Lifecycle]].

## Signed and serialized transactions

Alternatively, `cardano-wallet` allows clients to submit already signed and serialized transactions as a raw bytes blob. This can be done by submitting such serialized data as an `application/octet-stream` :

[`POST /proxy/transactions`](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/postExternalTransaction)

> In this scenario, the server engine will verify that the transaction is structurally well-formed and forward it to its associated node. If the transaction belongs to a known wallet, it will eventually show up in the wallet your wallet.

Reviewer: ```suggestion
In this scenario, the server engine will verify that the transaction is structurally well-formed and forward it to the node instance associated with it. If the transaction belongs to a known wallet, it will eventually show up in the wallet.
```
